src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/KotlinJvmTaskExecutor.kt

```diff
@@ -60,26 +60,29 @@ class KotlinJvmTaskExecutor @Inject internal constructor(
         sequenceOf(
             runCatching {
               context.execute("kotlinc") {
-                compileKotlin(context,
+                if (compileKotlin) {
+                  compileKotlin(context,
                     compiler,
                     args = baseArgs()
-                        .given(outputs.jar).notEmpty {
-                          append(codeGenArgs())
-                        }.given(outputs.abijar).notEmpty {
-                          plugin(abiPlugins.jvmAbiGen) {
-                            flag("outputDir", directories.classes)
-                          }
-                          given(outputs.jar).empty {
-                            plugin(abiPlugins.skipCodeGen)
-                          }
-                        },
+                      .append(codeGenArgs())
+                      .given(outputs.abijar).notEmpty {
+                        plugin(abiPlugins.jvmAbiGen) {
+                          flag("outputDir", directories.abiClasses)
+                        }
+                        given(outputs.jar).empty {
+                          plugin(abiPlugins.skipCodeGen)
+                        }
+                      },
                     printOnFail = false
-                )
+                  )
+                } else {
+                  emptyList()
+                }
               }
             },
             runCatching {
               context.execute("javac") {
-                if (outputs.jar.isNotEmpty()) {
```

Comment:
I think we can remove the Java compiler stuff from the Kotlin Builder but I'm not sure if you really want to keep it for backwards compatibility / safety net / preserve monolithic way as the default and have this PR behind a flag or something..... all this works nicely on Snaps Android codebase so I'm reasonably confident in it.

Reply:
Stepwise. We need to put these changes behind a flag.

Reply:
Sounds good. I'll do that this weekend.

Note that now Skylark supports build options, how do you feel if I create a Skylark Option for this or prefer an option in the tool chain like you did for experimental_abi_jars?

How about
`experimental_always_use_java_builders`
or
`experimental_new_action_flow`
or.... blanking on good names.

Skylark build flags: see: https://docs.google.com/document/d/1vc8v-kXjvgZOdQdnxPTaV0rrLxtP2XwnD2tAZlYJOqw/edit#heading=h.pn8nbhno42p5

Reply:
Added an experimental flag to use Java builders.

Reply:
I'm leaning towards toolchain support. The build flags are great to have, but command line options tend to proliferate and be non-discoverable.

Reply:
Added an option `experimental_use_java_builder` to the tool chain to enable the JavaBuilder flow.